diff --git a/types/svelte.d.ts b/types/svelte.d.ts
index 759e0f8acd94695eb8908a6a0c7d91dabf128055..826095eba2fd77e1a7d7ccff7e0cbfafff8a0cee 100644
--- a/types/svelte.d.ts
+++ b/types/svelte.d.ts
@@ -1,8 +1,11 @@
 declare module 'virtual:icons/*' {
-  export { SvelteComponentDev as default } from 'svelte/internal'
+  import { SvelteComponent } from 'svelte'
+  import type { SvelteHTMLElements } from 'svelte/elements'
+  export default class extends SvelteComponent<SvelteHTMLElements['svg']> {}
 }
 
 declare module '~icons/*' {
-  import { SvelteComponentTyped } from 'svelte'
-  export default class extends SvelteComponentTyped<svelte.JSX.IntrinsicElements['svg']> {}
+  import { SvelteComponent } from 'svelte'
+  import type { SvelteHTMLElements } from 'svelte/elements'
+  export default class extends SvelteComponent<SvelteHTMLElements['svg']> {}
 }
